### PR TITLE
Network Graph layout - debug window improvement. 

### PR DESCRIPTION
**Pull request by RandyMcMillan**
#### Rearrange the views for (imo) better layout and debug window resizing options.

Screen Shot 2021-04-23 at 6 29 30 PM

Removing "lblGraphRange" minimumSize enables the view to be reduced to the size of a desktop "widget" :)

https://github.com/bitcoin-core/gui/pull/291/files#diff-a24601363160c5ffd048f45a763e702c988b067f96e48a816c36f855f9820826L681

Screen Shot 2021-04-23 at 6 28 00 PM

The peers tab resizes if the detail view is revealed.

Screen Shot 2021-04-23 at 6 29 15 PM

Screen Shot 2021-04-23 at 6 29 21 PM

Info tab still useful at the "widget" size.

Screen Shot 2021-04-23 at 6 28 11 PM

Console tab still functional at "widget" size.

Screen Shot 2021-04-23 at 6 29 04 PM

### CODE REVIEW

**Comment by RandyMcMillan** [comment]
Tested with #290 #256

https://github.com/RandyMcMillan/gui/commits/net-graph-layout-290-256

**Comment by RandyMcMillan** [comment]
Tested with #271 

https://github.com/RandyMcMillan/gui/commits/net-graph-layout-271

**Comment by Bosch-0** [comment]
tACK https://github.com/bitcoin-core/gui/pull/291/commits/99f6f43bdd05ed83fca462c0514d6157bad1025b on 

Windows 10. 

Looks good to me, agree this does look slightly more clean. Other windows seem fine. Before / After screens: 

Frame 35

File: src/qt/forms/debugwindow.ui [hunk 1]
```diff
@@ -657,6 +657,181 @@
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
+           <item>
+            <widget class="QGroupBox" name="groupBox_2">
+             <property name="toolTip">
+              <string extracomment="Totals"/>
```
Comment (jonatack) [inline src/qt/forms/debugwindow.ui:663]: How does a user see this tooltip?

**Review by jonatack — COMMENTED** [review]
Tested Concept ACK. This seems like a nice design improvement.

Suggest reviewers look at the diff with `colorMoved = dimmed-zebra` and `colorMovedWs = allow-indentation-change`.

Is the slider working correctly? It seems like I lose the traffic history if I move it.

File: src/qt/forms/debugwindow.ui (hunk 1 above)
Comment (RandyMcMillan) [reply src/qt/forms/debugwindow.ui]: thanks - removed

Comment (RandyMcMillan) [reply src/qt/forms/debugwindow.ui]: The traffic history always took time to repaint.
I doubt there is an easy fix - it would require caching excessive data in anticipation of a resolution change.
I would lean toward prudent disk space usage over this inconvenience.

Comment (jonatack) [reply src/qt/forms/debugwindow.ui]: Thanks, makes sense and is unrelated to this change. Will re-review tomorrow.

**Comment by Talkless** [comment]
Idunno about Network Info tab. It's now inconsistent with Peers tab, where details are shown on the right.
What if we get more Network metrics in the future, not just received/sent? Keeping right pane allows for future growth.

**Comment by RandyMcMillan** [comment]
there is plenty of room for more tabs. 😄

**Comment by jonatack** [comment]
ACK 1cdcdb30ce57ab3847a839dcb71570c2c5ef0e36

**Comment by Talkless** [comment]
> there is plenty of room for more tabs. smile

What I mean is if there's more data series needed (in addition to "Sent", and "Received") on the _same_ graph, not about new tabs.

**Review by jarolrod — COMMENTED** [review]
tACK 1cdcdb30ce57ab3847a839dcb71570c2c5ef0e36

Screen Shot 2021-04-30 at 12 52 39 AM


Tested on macOS 11.3 Qt 5.15.2. I agree that this is a better use of space.

@RandyMcMillan 
You may want to fix your commit author; it's currently authored by `git`. You may want to change that back to you.

@Talkless 
> What I mean is if there's more data series needed (in addition to "Sent", and "Received") on the same graph, not about new tabs.

That can be figured out when new data series are proposed

**Comment by hebasto** [comment]
Concept ACK. IIUC this is a reincarnation of #90.

```
$ git log -1
commit 1cdcdb30ce57ab3847a839dcb71570c2c5ef0e36 (HEAD -> pr291-0430)
Author: git <git@gits-Air.deepspace.dynalias.org>
Date:   Mon Sep 14 17:21:44 2020 -0400

    Network Graph layout
```
^ this (the authorship) should be fixed before merging.

File: src/qt/forms/debugwindow.ui [hunk 2]
```diff
@@ -678,14 +850,8 @@
...
-             <property name="minimumSize">
-              <size>
-               <width>100</width>
-               <height>0</height>
-              </size>
-             </property>
-             <property name="alignment">
-              <set>Qt::AlignCenter</set>
+             <property name="text">
+              <string>Range</string>
```
Comment (hebasto) [inline src/qt/forms/debugwindow.ui:854]: Why this change?

This label will be overwritten in https://github.com/bitcoin-core/gui/blob/480bf01c295527bd212964efe4df3bb886db5654/src/qt/rpcconsole.cpp#L1006

and it is unnecessary burden to translate the "Range" string.

**Comment by hebasto** [comment]
`horizontalLayout_3` is useless:

Screenshot from 2021-04-30 14-45-45

With this change
```diff
--- a/src/qt/forms/debugwindow.ui
+++ b/src/qt/forms/debugwindow.ui
@@ -642,9 +642,7 @@
       <attribute name="title">
        <string>&amp;Network Traffic</string>
       </attribute>
-      <layout class="QHBoxLayout" name="horizontalLayout_3">
-       <item>
-        <layout class="QVBoxLayout" name="verticalLayout_4">
+      <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="TrafficGraphWidget" name="trafficGraph" native="true">
            <property name="sizePolicy">
@@ -867,8 +865,6 @@
            </item>
           </layout>
          </item>
-        </layout>
-       </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_peers">
```

we have

Screenshot from 2021-04-30 14-51-15

**Review by hebasto — COMMENTED** [review]
Approach ACK 0174e68fd59460b53a187bf396cb54e523185eb1

A slider glitching was introduced:
gui291-0430-1

I think the root of this bug is changed layout of the `sldGraphRange`, `lblGraphRange` and `btnClearTrafficGraph` widgets.

On master:
```xml
            <widget class="QLabel" name="lblGraphRange">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
```

This PR:
```xml
            <widget class="QLabel" name="lblGraphRange">
            </widget>
```

**Comment by RandyMcMillan** [comment]
Great catch! In #90 the peers tab toggle detailView wasn't merged yet! So this wasn't an issue.
Re-adding the properties now.

NOTE: I would have reused and rebased PR #90 - For some reason gh wouldn't let me reopen it.

**Comment by Talkless** [comment]
> > What I mean is if there's more data series needed (in addition to "Sent", and "Received") on the same graph, not about new tabs.
> 
> That can be figured out when new data series are proposed

So we gonna move back to the right? :) . Maybe adding a splitter, making right pane collapsible would help if someone wants "full screen like" graph?

**Comment by RandyMcMillan** [comment]
yes @Talkless - that is an option. I will take a look at that.

**Comment by RandyMcMillan** [comment]
Update: 797b1a5

Landscape Layout:

Screen Shot 2021-05-16 at 3 32 09 PM

Widget Layout: 

Screen Shot 2021-05-16 at 3 32 00 PM

File: src/qt/forms/debugwindow.ui [hunk 3]
```diff
@@ -871,22 +824,57 @@
...
-               </size>
-              </property>
-             </spacer>
-            </item>
            </layout>
           </widget>
          </item>
+         <item>
+          <widget class="QSlider" name="sldGraphRange">
+           <property name="minimumSize">
```
Comment (RandyMcMillan) [inline src/qt/forms/debugwindow.ui:832]: (re) setting minimum size to address sizing issue when time interval changes

File: src/qt/forms/debugwindow.ui [hunk 4]
```diff
@@ -871,22 +824,57 @@
...
+            <number>6</number>
+           </property>
+           <property name="orientation">
+            <enum>Qt::Horizontal</enum>
+           </property>
+          </widget>
+         </item>
+         <item>
+          <widget class="QLabel" name="lblGraphRange">
+           <property name="minimumSize">
```
Comment (RandyMcMillan) [inline src/qt/forms/debugwindow.ui:857]: (re) setting minimum size to address sizing issue when time interval changes

**Comment by RandyMcMillan** [comment]
Slider Issue Fixed:

slider

**Review by ShaMan239 — COMMENTED** [review]
Concept ACK
Tested Successfully on Ubuntu 20.04

The New Layout for the Network Traffic Window looks much better than the current master.
It is also not affecting the functionality of other Node Windows.
Nice Work!
The bug discussed [here](https://github.com/bitcoin-core/gui/pull/291#pullrequestreview-649577005) is also fixed. This was done by op by setting a minimum size of 60px for the lblGraphRange.
But I think op should increase the minimum size a little more because it is causing some overflow issues. Adding a screenshot of the problem:
Screenshot from 2021-08-07 01-06-56(1)
Overflowing of time (in the blue box near the slider)

Other than this the PR seems to work excellently
Adding some screenshots for comparison.

**On Maximum Size:**
|Master|PR|
|---------|----|
|Screenshot from 2021-08-07 00-11-07|Screenshot from 2021-08-07 00-10-02|

**On Minimum Size:**

|Master|PR|
|---------|----|
|Screenshot from 2021-08-07 00-11-00(2)|Screenshot from 2021-08-07 00-09-48|

**Difference in Layout (Screenshot from Qt Designer):**

|Master|PR|
|---------|----|
|Screenshot from 2021-08-07 00-52-00|Screenshot from 2021-08-07 00-47-58|
<p>

**Comment by hebasto** [comment]
I've reconsidered my opinion, and withdraw my ACKs (https://github.com/bitcoin-core/gui/pull/291#issuecomment-830029755, https://github.com/bitcoin-core/gui/pull/291#pullrequestreview-649577005).

Agree with @Talkless's point (https://github.com/bitcoin-core/gui/pull/291#issuecomment-826322546, https://github.com/bitcoin-core/gui/pull/291#issuecomment-827027675, https://github.com/bitcoin-core/gui/pull/291#issuecomment-831420253).

One of such changes, that require the current layout, are "Received" and "Send" counters separated by data type: blocks, transactions, addresses, service messages. Especially, this could be helpful with upcoming Erlay protocol (at least for me, of course).

So, for now, Concept NACK from me.

**Comment by rebroad** [comment]
Can we make it so the default is 3 hours not 30 minutes? Or make it so that this can be configured somewhere?

Also, can we make it so that when the time range is changed that the data is not lost?

File: src/qt/forms/debugwindow.ui [hunk 5]
```diff
@@ -642,79 +642,32 @@
...
+            <string/>
            </property>
-           <layout class="QVBoxLayout" name="verticalLayout_5">
+           <property name="flat">
+            <bool>true</bool>
+           </property>
+           <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
-             <layout class="QHBoxLayout" name="horizontalLayout_4">
+             <layout class="QHBoxLayout" name="RecievedBox">
```
Comment (luke-jr) [inline src/qt/forms/debugwindow.ui:668]: typo here and 2 more

**Comment by RandyMcMillan** [comment]
Looks good with PR #473 as well...

Screen Shot 2021-11-20 at 7 39 55 PM

**Comment by RandyMcMillan** [comment]
Minimum size presentation: 

Screen Shot 2021-11-20 at 7 41 06 PM

**Comment by rebroad** [comment]
> Minimum size presentation:
> 
> Screen Shot 2021-11-20 at 7 41 06 PM

I like this new format - also, it makes it more possible to add a button above/below reset which could be used to toggle between linear and non-linear (rather than clicking on the graph).
